### ISSUE TITLE
Resource usage %% metrics do not work if containers do not specify requests/limits

### ISSUE THREAD

**Issue by wasker** [Azure/prometheus-collector#420]
For example, Prometheus doesn't specify requests/limits for most of its containers. This makes dashboards plotting CPU/memory usage %% less useful/misleading, because only e.g., config-reloader is visible. In this case, pod metrics (sum of all containers) is even stranger, because the whole Prometheus' resource usage is divided by requests/limits of only config-reloader.

**Comment by gracewehner** [comment]
Hi @wasker, thanks for opening an issue. Could you please give more details about which metrics and dashboards you are using? The dashboards used in our Managed Prometheus offering are the same as the community kubernetes ones. I see in the Kubernetes/Compute Resource dashboards the metric `container_cpu_usage_seconds_total` is graphed separately from `kube_pod_container_resource_requests` and `kube_pod_container_resource_limits` so that the utilization still shows for containers without requests/limits.

**Comment by wasker** [comment]
Anything that's defined [like this](https://github.com/Azure/prometheus-collector/blob/d84a02b63b6650f6a54bf1760eb4160dfe49b92b/mixins/kubernetes/dashboards/resources/pod.libsonnet#L180).

The problem that I'm trying to solve is that container_cpu_usage_seconds_total:sum_irate (or more specifically windows_container_cpu_usage_seconds_total:sum_rate for us) crosses 100% threshold. I can't find any explanation to this behavior, but dividing sum_rate by limits seem to at least match what we observe in HPA metrics. However, this breaks down for containers/pods that don't have limits configured.

**Comment by wasker** [comment]
For example, I ran `seq 3 | xargs -P0 -n1 md5sum /dev/zero` in the prometheus container. The only CPU metric I can use is `node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate`. That metric evaluates to 3.03 in my experiment (to compare, other containers evaluate same metric to values << 1).

How do I reason about it?

**Comment by wasker** [comment]
Well, yes, I get that I can find some utilization number somewhere. But don't you find this counterintuitive/confusing that the user has to be aware that containers w/o requests/limits do not show up in one place and when they're looking in another place, the visualization is different (% vs. cores)?

**Comment by wasker** [comment]
@vishiy I do not expect anything from anyone. My initial question was asked as part of my research on how to improve resource utilization monitoring on clusters that my team operates.

I found this repo by looking for a solution of the problem when we couldn't get consistent CPU %% usage numbers and raised this issue to see if I was missing something.

Apparently, no, I didn't miss anything. The behavior when cluster operator has to look in different places to figure out CPU usage for different kind of containers, seems to be an accepted practice and everyone is OK with that.

I was not happy with this behavior, so I solved the problem for our team (yes, by using node capacity as a denominator in such cases). Seems to be a very quick, but effective fix, which nicely improved our monitoring experience.

I believe that there's an opportunity to improve dashboards that you're offering out of the box by providing a coherent performance picture of the monitored cluster, hence my last comment. Make of it what you will.
